tests/integration/credset_test.go

```diff
@@ -0,0 +1,516 @@
+		It("should run porter apply", func() {
+			By("creating an agent action", func() {
+				ctx := context.Background()
+				ns := createTestNamespace(ctx)
+				name := "cs-update-" + ns
+
+				Log(fmt.Sprintf("create credential set '%s' for credset", name))
+				cs := NewTestCredSet(name)
+				cs.Spec.Name = "cs-update-test"
+				cs.ObjectMeta.Namespace = ns
+				cred := porterv1.Credential{
+					Name: "test-credential",
+					Source: porterv1.CredentialSource{
+						Secret: name,
+					},
+				}
+				cs.Spec.Credentials = append(cs.Spec.Credentials, cred)
+				cs.Spec.Namespace = ns
+
+				Expect(k8sClient.Create(ctx, cs)).Should(Succeed())
+				Expect(waitForPorterCS(ctx, cs, "waiting for credential set to apply")).Should(Succeed())
+				validateCredSetConditions(cs)
+
+				Log("verify it's created")
+				jsonOut := runAgentAction(ctx, "create-check-credentials-list", ns, []string{"credentials", "list", "-n", ns, "-o", "json"})
+				firstName := gjson.Get(jsonOut, "0.name").String()
+				numCredSets := gjson.Get(jsonOut, "#").Int()
+				numCreds := gjson.Get(jsonOut, "0.credentials.#").Int()
+				firstCredName := gjson.Get(jsonOut, "0.credentials.0.name").String()
+				Expect(int64(1)).To(Equal(numCredSets))
+				Expect(int64(1)).To(Equal(numCreds))
+				Expect(cs.Spec.Name).To(Equal(firstName))
+				Expect("test-credential").To(Equal(firstCredName))
+
+				Log("update a credential set")
+				newSecret := "update-credset"
+				newCred := porterv1.Credential{
+					Name: "new-credential",
+					Source: porterv1.CredentialSource{
+						Secret: newSecret,
+					},
+				}
+				cs.Spec.Credentials = append(cs.Spec.Credentials, newCred)
+				patchCS := func(cs *porterv1.CredentialSet) {
+					controllers.PatchObjectWithRetry(ctx, logr.Discard(), k8sClient, k8sClient.Patch, cs, func() client.Object {
+						return &porterv1.CredentialSet{}
+					})
+				}
+				patchCS(cs)
+				Expect(waitForPorterCS(ctx, cs, "waiting for credential update to apply")).Should(Succeed())
+				Log("verify it's updated")
+				jsonOut = runAgentAction(ctx, "update-check-credentials-list", ns, []string{"credentials", "list", "-n", ns, "-o", "json"})
+				updatedFirstName := gjson.Get(jsonOut, "0.name").String()
+				updatedNumCredSets := gjson.Get(jsonOut, "#").Int()
+				updatedNumCreds := gjson.Get(jsonOut, "0.credentials.#").Int()
+				updatedFirstCredName := gjson.Get(jsonOut, "0.credentials.0.name").String()
+				updatedSecondCredName := gjson.Get(jsonOut, "0.credentials.1.name").String()
+				Expect(int64(1)).To(Equal(updatedNumCredSets))
+				Expect(int64(2)).To(Equal(updatedNumCreds))
+				Expect(cs.Spec.Name).To(Equal(updatedFirstName))
+				Expect("test-credential").To(Equal(updatedFirstCredName))
+				Expect("new-credential").To(Equal(updatedSecondCredName))
+			})
+		})
+	})
+})
+
+var _ = Describe("CredentialSet delete", func() {
+	Context("when an existing CredentialSet is delete", func() {
+		It("should run porter credentials delete", func() {
```

Comment:
This test seems to be a copy of the update test above, with the extra delete action and versification. One way to make this easier to maintain going forward would be to combine them into a single test that walks through the entire lifecycle of a credential set.